libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -8,24 +8,48 @@ import {
 } from '@tmo/books/data-access';
 import { FormBuilder } from '@angular/forms';
 import { Book } from '@tmo/shared/models';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  takeUntil
+} from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
   styleUrls: ['./book-search.component.scss']
 })
-export class BookSearchComponent {
+export class BookSearchComponent implements OnInit, OnDestroy {
   books$ = this.store.select(getAllBooks);
 
   searchForm = this.fb.group({
     term: ''
   });
 
+  unsubscribeSubject$ = new Subject();
+
   constructor(
     private readonly store: Store,
     private readonly fb: FormBuilder
   ) {}
 
+  ngOnInit(): void {
+    this.searchForm
+      .get('term')
+      .valueChanges.pipe(
+        debounceTime(500),
+        distinctUntilChanged(),
+        takeUntil(this.unsubscribeSubject$)
```

Comment:
You have used host listener to unsubscribe observables so question is, do we really need this ```takeUntil(this.unsubscribeSubject$)```?

Reply:
```
searchExample(): void {
    this.searchForm.controls.term.setValue('javascript');
    this.searchBooks();
  }
```

As we implemented instant search, do we really need ``` this.searchBooks();``` statement in above method?